Realm/RLMObject.mm

```diff
@@ -40,7 +40,11 @@ - (instancetype)init {
 }
 
 - (instancetype)initWithValue:(id)value {
-    return [super initWithValue:value schema:RLMSchema.sharedSchema];
+    RLMSchema *schema = RLMSchema.partialSharedSchema;
+    if (![schema schemaForClassName:self.class.className]) {
```

Comment:
They're only cached in the partial & full shared schemas. What I'm proposing is finer schema caching like a mutable (and private) `+[RLMSchema parsedSchemas]` so that instead of loading the entire shared schema if a single class is missing from the partial shared schema, the missing object schema is adding to the parsed schemas and *that*s passed in to `initWithValue:schema:`.

This would further minimize memory usage from loaded object schemas.